src/cf.data.pre

```diff
@@ -7007,14 +7007,12 @@ DOC_END
 
 NAME: esi_parser
 IFDEF: USE_SQUID_ESI
-COMMENT: libxml2|expat|custom
+COMMENT: libxml2|expat
 TYPE: string
 LOC: ESIParser::Type
-DEFAULT: custom
+DEFAULT: libxml2
```

Comment:
What would happen when Squid is built with libexpat and without libxml2? Only one of those two libraries is required to get USE_SQUID_ESI enabled, right? It feels like the default should be influenced by the library availability.

Reply:
Squid can use either or both. The code seems to prefer libxml2 and a quick check online shows that library is a bit more active than libexpat. Though its documentation is not as good. They both have packages or binaries available for all the OS we formally support, and then some.

Reply:
Sorry, missed your meaning for a bit there. Yes expat-only would be a startup problem. Fixing that now by moving the default selection into esi/Parser.cc based on the HAVE_* instead of static value in cf.data.pre